ast/compile.go

```diff
@@ -3113,7 +3113,24 @@ func declaredVars(x interface{}) VarSet {
 				})
 			} else if decl, ok := x.Terms.(*SomeDecl); ok {
 				for i := range decl.Symbols {
-					vars.Add(decl.Symbols[i].Value.(Var))
+					switch val := decl.Symbols[i].Value.(type) {
+					case Var:
+						vars.Add(val)
+					case Call:
```

Comment:
⚠️ This was added for completeness, but I have yet to find the right tests to adapt to cover this.

Reply:
Yup thanks. Added what was necessary there, we're not resolving the references in the last argument position properly, too.